test/libsolidity/SemanticTest.cpp

```diff
@@ -99,6 +99,10 @@ SemanticTest::SemanticTest(
 	if (m_runWithABIEncoderV1Only && !solidity::test::CommonOptions::get().useABIEncoderV1)
 		m_shouldRun = false;
 
+	// Sanity check
+	if (m_runWithABIEncoderV1Only && (compileViaYul == "true" || compileViaYul == "also"))
+		BOOST_THROW_EXCEPTION(runtime_error("ABIEncoderV1Only can not be used with compileViaYul=" + compileViaYul + ", set it to false or omit the flag."));
```

Comment:
Might be a good idea to say why. I.e. that `abicoder v1` pragma is just ignored in Yul mode and everything actually runs as if v2 was used.

Reply:
Fixed.